File: app/src/test/java/com/main/climbingdiary/common/GradeConverterTest.java
package com.main.climbingdiary.common;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;


public class GradeConverterTest {


    private static Stream<Arguments> provideStringsForTest() {
        return Stream.of(
                Arguments.of("IX+/X-", GradeConverter.convertFrenchToUiaa("8a")),
                Arguments.of(null, GradeConverter.convertFrenchToUiaa("22")),
                Arguments.of("6a", GradeConverter.convertUiaaToFrench("VI+")),
                Arguments.of(null, GradeConverter.convertUiaaToFrench("test")),
                Arguments.of("7c", GradeConverter.convertUiaaToFrench("IX")),
                Arguments.of("5c", GradeConverter.convertUiaaToFrench("VI")),
                Arguments.of("VIII+", GradeConverter.convertFrenchToUiaa("7a+"))
        );
    }

    @ParameterizedTest
    @MethodSource("provideStringsForTest")
    public void testGradeConverter(String input, String expected) {
        assertEquals(input, expected);
    }

}

File: app/src/test/java/com/main/climbingdiary/common/GradeConverterTest.kt
package com.main.climbingdiary.common

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class GradeConverterTest {

    @ParameterizedTest
    @CsvSource("VIII-,6c+", "IX+/X-,8a", "IX-,7b+")
    fun convertUiaaToFrench(input: String, expected: String) {
        val actualValue = GradeConverter.convertUiaaToFrench(input)
        assertEquals(expected, actualValue)
    }

    @ParameterizedTest
    @CsvSource("VIII-,6c+", "IX+/X-,8a", "IX-,7b+")
    fun convertFrenchToUiaa(input: String, expected: String) {
        val actualValue = GradeConverter.convertFrenchToUiaa(expected)
        assertEquals(input, actualValue)
    }

    @ParameterizedTest
    @CsvSource("8+/9-, 7b", "9+, 7c+", "7+/8-, 6c")
    fun convertStringToCorrectScale(input: String, expected: String) {
        val actualValue = GradeConverter.convertAnyToFrench(input)
        assertEquals(actualValue, expected)
    }
}